crypto/sharing.go

```diff
@@ -173,6 +173,22 @@ func (mach *OlmMachine) receiveSecret(ctx context.Context, evt *DecryptedOlmEven
 		return
 	}
 
+	// https://spec.matrix.org/v1.10/client-server-api/#msecretsend
+	// "The recipient must ensure... that the device is a verified device owned by the recipient"
+	if senderDevice, err := mach.GetOrFetchDevice(ctx, evt.Sender, evt.SenderDevice); err != nil {
```

Comment:
Hrm, should I GetOrFetchDevice here or just look in the local store?

The key backup signature checking only looks in the local store
https://github.com/mautrix/go/blob/100d945d3977a3261e01090e245c9ffa3bd47af2/crypto/keybackup.go#L84

Reply:
It should always have the list cached, so shouldn't make a difference either way. If your own device list is somehow missing, it's probably only good that /sync is blocked until the list is fetched